conans/client/toolchain/cmake.py

```diff
@@ -0,0 +1,185 @@
+        #   CMAKE_CXX_FLAGS. See https://github.com/android/ndk/issues/323
+        if(CONAN_TOOLCHAIN_INCLUDED)
+          return()
+        endif()
+        set(CONAN_TOOLCHAIN_INCLUDED true)
+
+        message("Using Conan toolchain through ${CMAKE_TOOLCHAIN_FILE}.")
+        
+        # CMAKE_BUILD_TYPE: Use it only if it isn't a multi-config generator
+        get_property(_GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG )
+        if(NOT _GENERATOR_IS_MULTI_CONFIG)
+            set(CMAKE_BUILD_TYPE "{{ CMAKE_BUILD_TYPE }}" CACHE STRING "Choose the type of build." FORCE)
+        endif()
+        unset(_GENERATOR_IS_MULTI_CONFIG)
+
+        # Configure
+        # -- CMake::command_line
+        {% if generator_platform %}set(CMAKE_GENERATOR_PLATFORM "{{ generator_platform }}"){% endif %}
+        {% if toolset %}set(CMAKE_GENERATOR_TOOLSET "{{ toolset }}"){% endif%}
+        
+        # --  - CMake.flags --> CMakeDefinitionsBuilder::get_definitions
+        {%- for it, value in definitions.items() %}
+        set({{ it }} "{{ value }}" CACHE STRING "Do we want to set all these vars in the cache?" FORCE)
+        {%- endfor %}
+        
+        # Set some environment variables
+        {%- for it, value in environment.items() %}
+        set(ENV{{ '{' }}{{ it }}{{ '}' }} "{{ value }}")
+        {%- endfor %}
+        
+        include("{{ conan_adjustements_cmake }}")
+        
+        get_property( _CMAKE_IN_TRY_COMPILE GLOBAL PROPERTY IN_TRY_COMPILE )
+        if(NOT _CMAKE_IN_TRY_COMPILE)
+            message(">>>> NOT TRY COMPILE")
+            set(CMAKE_PROJECT_INCLUDE "{{ conan_project_include_cmake }}")  # Will be executed after the project
+            include("{{ conan_adjustements_cmake }}")
+
+            # We are going to adjust automagically many things as requested by Conan
+            #   these are the things done by 'conan_basic_setup()'
+            
+            conan_set_flags("")
+            conan_set_flags("_RELEASE")
+            conan_set_flags("_DEBUG")
+            # TODO: Do we want to set flags here if we are going to use targets? Maybe these flags are defined inside the 'toolchain' method of the recipe
+            
+            conan_set_rpath()
+            conan_set_std()
+            conan_set_fpic()
+            
+            #conan_check_compiler()
+            conan_set_libcxx()
+            #conan_set_vs_runtime()  # This goes to the after-project() command call
+            conan_set_find_paths()
+            conan_set_find_library_paths()
+            
+            set(CMAKE_CXX_FLAGS_INIT "${CONAN_CXX_FLAGS}" CACHE STRING "" FORCE)
+            set(CMAKE_C_FLAGS_INIT "${CONAN_C_FLAGS}" CACHE STRING "" FORCE)
+            set(CMAKE_SHARED_LINKER_FLAGS_INIT "${CONAN_SHARED_LINKER_FLAGS}" CACHE STRING "" FORCE)
+            set(CMAKE_EXE_LINKER_FLAGS_INIT "${CONAN_EXE_LINKER_FLAGS}" CACHE STRING "" FORCE)
```

Comment:
Right now the toolchain is generated per configuration (`conan install <...> --profile XXX`) and it contains many things just for that single configuration. It would be the same to have those flags assigned to the `CMAKE_CXX_FLAGS_XXX_INIT` or to the main `CMAKE_CXX_FLAGS_INIT`. Also, the user can conditionally add definitions depending on the configuration... 

The only way to get the valid elements (flags, defs by user,...) for release and debug would be to run the toolchain function with both configurations and mix them. Probably it is something we don't want to do.

Maybe there is a better and easier way, but I can't see it.

Reply:
This is still probably the biggest thing to address: how to use the toolchain feature with VS IDE switching Debug/Release configurations. I think this is still necessary, but it seems this has been improved lately? Is it now supported, isn't it?